index.js

```diff
@@ -356,6 +356,31 @@
     ($.Undefined  = type0('Undefined'))
   ];
 
+  //  ValidNumber :: Type
+  $.ValidNumber = $.NullaryType(
+    'sanctuary-def/ValidNumber',
+    function(x) { return $.Number.test(x) && !isNaN(x); }
+  );
+
+  //  FiniteNumber :: Type
+  $.FiniteNumber = $.NullaryType(
+    'sanctuary-def/FiniteNumber',
+    function(x) { return $.ValidNumber.test(x) && x !== Infinity &&
+                                                  x !== -Infinity; }
+  );
+
+  //  NonZeroValidNumber :: Type
+  $.NonZeroValidNumber = $.NullaryType(
+    'sanctuary-def/NonZeroValidNumber',
+    function(x) { return $.ValidNumber.test(x) && x !== 0 && x !== -0; }
```

Comment:
> Or we could jusy use `==`.

I'm happy to use `== 0` if you prefer it, since in this context we can be sure both operands will be of the same type. You'll just need to relax the linting rules.